tests/server.rs

```diff
@@ -0,0 +1,125 @@
+// Copied from https://github.com/seanmonstar/reqwest/blob/ab49de875ec2326abf25f52f54b249a28e43b69c/tests/support/server.rs
+// with some slight tweaks
+use std::convert::Infallible;
+use std::future::Future;
+use std::net;
+use std::sync::mpsc as std_mpsc;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+
+use hyper::service::{make_service_fn, service_fn};
+use tokio::runtime;
+use tokio::sync::oneshot;
+
+pub struct Server {
+    addr: net::SocketAddr,
+    hits: Arc<Mutex<u8>>,
+    // The panic_rx channel is to make sure the test fails if the server panics.
+    // If the server panics, the message is not sent and the recv call panics.
```

Comment:
https://discord.com/channels/500028886025895936/670880858630258689/906845871021383700

Reply:
I'm not in that server (and can't tell which server it is) so I can't view that message

Reply:
It is in [tokio's discord server](https://discord.com/invite/tokio)